### DISCUSSION TITLE
How to accept input from ^find?

### DISCUSSION THREAD

**Discussion by jiripospisil** [nushell/nushell#11257] [Q&A]
Hello, I would like to use `^find` to get the initial set of directory entries and further process them inside Nushell as usual. Basically I would like to be able to write something like this:

```
^find . -print0 | from find | get size
# or
^find . -print0 | into file | get size
```

I cannot use the built-in `ls/glob` because they support only basic filters. Is there a way to do that?

**Comment by fdncred** [comment]
I'm confused as to what `^find . --print0` does that `glob **/*` doesn't do.

**Reply by jiripospisil** [reply]
As far as I know it for example doesn't support filtering based on file permissions (and many other filters and expressions, see `man find`). That's why I wrote `...because they support only basic filters`. Maybe I should have written the question more like "accept an existing list of entries into the pipeline" rather than tying it specifically to `^find`.

**Reply by fdncred** [reply]
Typically, if you want to integrate some external tool's output into nushell tables, you need to parse the output. Currently, this looks something like this: `^external | lines | parsing_command` where `parsing_command` is usually one of these 3.
- parse
- from ssv
- detect columns

it could also look like `^external | lines | each | parsing_command`.

**Comment by leiserfg** [comment]
I think having a  `print0` format will be helpful to interact with programs that input/output items that way (usually things that can bue multiline). For instance `fzf --print0`

**Reply by halloleo** [reply]
Having read @weirdan's [comment](https://github.com/nushell/nushell/discussions/11257#discussioncomment-13474753) you could define a custom command:

    def "from print0" [] { str trim --char (char nul) | split row (char nul) }

Seem to work alright.

**Comment by weirdan** [comment]
```nushell
^find -name '*.nu' -print0 | str trim --char (char nul) | split row (char nul)
```
should do what you need.

**Reply by halloleo** [reply]
Nice one! A bit annoying that `str trim --char (char nul)` is needed - just for the last `nul`...
